### PR TITLE
Create more-for timer with action buttons

### LINKED ISSUE #3594: Add timer controls to more-info dialog
**Description of problem:**

There are four Timer services (`timer.cancel`, `timer.finish`, `timer.pause`, `timer.start`) but these are only accessible from Developer Tools -> Services tab currently. 

Not only does it feel a little bit silly having to go to "Developer Tools" to simply cancel a timer, but more importantly the permissions system blocks off that area to regular user accounts. So a non-admin account would have no way of stopping a nap timer if they woke up early for example.

It would be a lot more user friendly if these controls were displayed on the more-info dialog for timers. Here is a mockup:

timer-moreinfo

### PR DESCRIPTION
**Pull request by iantrich**
Closes https://github.com/home-assistant/home-assistant-polymer/issues/3594

Screen record from 2019-09-04 20 48 30@2x

### CODE REVIEW

File: src/dialogs/more-info/controls/more-info-timer.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+class MoreInfoTimer extends LitElement {
+  @property() public hass!: HomeAssistant;
+  @property() public stateObj?: any; // TODO
+
+  protected render(): TemplateResult | void {
+    if (!this.stateObj) {
+      return html``;
+    }
+
+    const actions = ["start", "pause", "cancel", "finish"];
```
Comment (balloob) [inline src/dialogs/more-info/controls/more-info-timer.ts]: Can we translate these ?

File: src/dialogs/more-info/controls/more-info-timer.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  property,
+  PropertyValues,
+} from "lit-element";
+import "@material/mwc-button";
+
+import { HomeAssistant } from "../../../types";
+
+class MoreInfoTimer extends LitElement {
+  @property() public hass!: HomeAssistant;
+  @property() public stateObj?: any; // TODO
```
Comment (balloob) [inline src/dialogs/more-info/controls/more-info-timer.ts]: Please add a TimerEntity type to `data/timer.ts`

File: src/dialogs/more-info/controls/more-info-timer.ts (hunk 1 above)
Comment (balloob) [reply src/dialogs/more-info/controls/more-info-timer.ts]: Surprised we didn't had timer translations for these yet.

File: src/dialogs/more-info/controls/more-info-timer.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  protected render(): TemplateResult | void {
+    if (!this.stateObj) {
+      return html``;
+    }
+
+    const actions = ["start", "pause", "cancel", "finish"];
+
+    return html`
+      <ha-attributes .stateObj="${this.stateObj}"></ha-attributes>
+      ${actions.map((state) => {
```
Comment (balloob) [inline src/dialogs/more-info/controls/more-info-timer.ts]: We should hide/show buttons based on the current state.

So only show a cancel if it has been started before.  Only show pause if it is running etc.

File: src/dialogs/more-info/controls/more-info-timer.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  protected updated(changedProps: PropertyValues) {
+    super.updated(changedProps);
+    if (!changedProps.has("stateObj") || !this.stateObj) {
+      return;
+    }
+  }
+
+  private _handleActionClick(e: MouseEvent): void {
+    const action = (e.currentTarget as any).action;
+    this.hass.callService("timer", action, undefined);
```
Comment (balloob) [inline src/dialogs/more-info/controls/more-info-timer.ts]: You are not targeting an entity?

File: src/dialogs/more-info/controls/more-info-timer.ts [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  }
+
+  static get styles(): CSSResult {
```
Comment (balloob) [inline src/dialogs/more-info/controls/more-info-timer.ts]: If we have no CSS, no need to include this.

File: src/dialogs/more-info/controls/more-info-timer.ts [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    return css``;
+  }
+}
+
+customElements.define("more-info-timer", MoreInfoTimer);
```
Comment (balloob) [inline src/dialogs/more-info/controls/more-info-timer.ts]: Use decorator.

**Comment by iantrich** [comment]
Sorry, I meant to put a WIP on this. Was just pusing it up before switching to something else.

**Review by balloob — APPROVED** [review]

File: src/dialogs/more-info/controls/more-info-timer.ts [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+
+  @property() public stateObj?: TimerEntity;
+
+  protected render(): TemplateResult | void {
+    if (!this.hass || !this.stateObj) {
+      return html``;
+    }
+
+    return html`
+      <ha-attributes .stateObj="${this.stateObj}"></ha-attributes>
```
Comment (balloob) [inline src/dialogs/more-info/controls/more-info-timer.ts]: Let's filter out `remaining`. It's confusing as it is not the actual remaining, but the remaining time when the state changed.

Comment (balloob) [reply src/dialogs/more-info/controls/more-info-timer.ts]: Also let's drop quotes around dynamic expressions
```suggestion
      <ha-attributes .stateObj=${this.stateObj}></ha-attributes>
```

**Pull request merged**